File: day-01/part-1/badouralix.py
from runners.python import Submission


class BadouralixSubmission(Submission):

	def run(self, s):
		# :param s: input in string format
		# :return: solution flag
		return sum([int(s[i]) for i in range(len(s)) if s[i-1] == s[i]])

File: day-04/part-2/badouralix.js

/**
* @param {{string}} s puzzle input in string format
* @returns solution flag
*/
function run(s) {
	return s.split('\n').filter(isValid).length;
}

function isValid(passphrase) {
	const set = new Set();
	passphrase.split(' ').forEach(word => {
		set.add(word.split('').sort().join(''));
	});
	return passphrase.split(' ').length == set.size;
}
